src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -153,17 +173,28 @@
    (fn [{:keys [state]}]
      (let [{:keys [methods loaded-config]} @state]
        (when (and (not methods) loaded-config)
-         (let [{:keys [methodName methodNamespace]} (get-in loaded-config [:methodConfiguration :methodRepoMethod])]
-           (endpoints/call-ajax-orch
-            {:endpoint (endpoints/list-method-snapshots methodNamespace methodName)
-             :on-done (fn [{:keys [success? get-parsed-response status-text]}]
-                        (let [response (get-parsed-response)]
-                          (if success?
-                            (swap! state assoc
-                                   :methods-response response
-                                   :methods {[methodNamespace methodName] (mapv :snapshotId response)})
-                            ;; FIXME: :error-message is unused
-                            (swap! state assoc :error-message status-text))))})))))
+         (let [{:keys [methodName methodNamespace]} (get-in loaded-config [:methodConfiguration :methodRepoMethod])
+               repo (get-in loaded-config [:methodConfiguration :methodRepoMethod :sourceRepo])]
+           (case repo
+             "agora" (endpoints/call-ajax-orch
+                      {:endpoint (endpoints/list-method-snapshots methodNamespace methodName)
+                       :on-done (fn [{:keys [success? get-parsed-response status-text]}]
+                                  (let [response (get-parsed-response)]
+                                    (if success?
+                                      (swap! state assoc
+                                             :methods-response response
+                                             :methods {[methodNamespace methodName] (mapv :snapshotId response)})
+                                      ;; FIXME: :error-message is unused
+                                      (swap! state assoc :error-message status-text))))})
+             "dockstore" (let [path (get-in loaded-config [:methodConfiguration :methodRepoMethod :methodPath])]
+                           (endpoints/dockstore-get-versions path
+                                                             (fn [{:keys [success? get-parsed-response status-text]}]
+                                                               (if success?
+                                                                 (swap! state assoc
+                                                                        :methods-response nil
+                                                                        ;; vector only used when checking redaction, N/A for Dockstore
+                                                                        :methods {[nil nil] (mapv :name (get-parsed-response))})
```

Comment:
this is very confusing, you have a map with a key that is a vector of `nil, nil`, whose value is a vector?

Reply:
I think I get why this is like this... but it makes me think that we shouldn't need to get the dockstore versions at all. Like the entire `"dockstore"` branch of the `case` isn't needed.